.github/workflows/author_approval.yml

```diff
@@ -0,0 +1,35 @@
+name: Author Approval Label
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  label:
+    runs-on: ubuntu-latest
+    # Run on comments that are:
+    # 1) on PRs, not issues,
+    # 2) not from bot users and,
+    # 3) include the string "approved"
+    # If so, we will do the work to check that the commenter is the package author,
+    # and conditionally apply the author-approved label.
+    if: ${{ github.event.issue.pull_request && github.event.issue.user.type != 'Bot' && contains(github.event.comment.body, 'approved') }}
+    steps:
+      - name: Verify package author
+        id: verify-author
+        env:
+          PR_BODY: ${{ github.event.issue.body }}
+          COMMENTER: ${{ github.event.comment.user.login }}
+        shell: julia --color=yes {0}
+        run: |
+          m = match(r"Created by: @([^\s]+)", ENV["PR_BODY"])
```

Comment:
Do we need to enforce that at the end of the `@username`, we have either `\n` or `$`?

Otherwise, wouldn't an approval by user `@foo` be able to approve a PR created by user `@foobar`?

Reply:
I think it's fine actually:
```julia
julia> r = r"Created by: @([^\s]+)"
r"Created by: @([^\s]+)"

julia> match(r, "Created by: @foobar")[1]
"foobar"

julia> match(r, "Created by: @foo")[1]
"foo"
```
i.e. we will match the whole username, whatever it is, and then we string-compare (not regex) against the commenter